### ISSUE TITLE
Type names in PostgreSQL CREATE TYPE are not considered to be type names

### ISSUE THREAD

**Issue by nene** [sqlfluff/sqlfluff#6792]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I gave the following code to SQL Fluff:

```sql
CREATE TYPE foo AS ENUM ('a', 'b', 'c');

CREATE TABLE t (name FOO);
```

### Expected Behaviour

I expected to see an error CP05, like "Datatypes must be consistently lower case".

### Observed Behaviour

Instead I received no errors. But clearly the naming of data types in the code is inconsistent. Type `foo` is defined in lowercase, but used in uppercase.

The same problem applies to `ALTER TYPE` and `DROP TYPE`.

At the same time `CREATE FUNCTION` is treated as expected. When I give SQL Fluff the code:

```sql
CREATE FUNCTION foo() AS '1 + 2';
SELECT FOO();
```

I get an error as expected: CP03 Function names must be consistently lower case.

I would expect all type names to be treated as type names regardless of context, similarly in how all function names are treated as function names.

### How to reproduce

Just paste the above code to [sqlfluff online](https://online.sqlfluff.com/fluffed?sql=H4sIAO9582cC_3MOcnUMcVUIiQxwVUjLz7fm5eLlcoaKOTr5uCqUKGhkpii4-fvrKOQl5qaCWJpAVQDjTJDPOAAAAA%3D%3D&dialect=PostgreSQL).

### Dialect

PostgreSQL

### Version

I don't know how to determine the version of [sqlfluff online](https://online.sqlfluff.com/).

### Configuration

None.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
